src/jit/compiler.hpp

```diff
@@ -1682,13 +1682,22 @@ inline unsigned Compiler::lvaGrabTemp(bool shortLifetime DEBUGARG(const char* re
         lvaTable    = newLvaTable;
     }
 
-    lvaTable[lvaCount].lvType    = TYP_UNDEF; // Initialize lvType, lvIsTemp and lvOnFrame
-    lvaTable[lvaCount].lvIsTemp  = shortLifetime;
-    lvaTable[lvaCount].lvOnFrame = true;
+    const unsigned tempNum = lvaCount;
+    lvaCount++;
 
-    unsigned tempNum = lvaCount;
+    lvaTable[tempNum].lvType    = TYP_UNDEF; // Initialize lvType, lvIsTemp and lvOnFrame
```

Comment:
nit: I think a comment that applies to multiple lines should be above the first line (though I realize this was already here).

Reply:
I'll fix this as part of #19345.